src/caffeinated_whale_cli/commands/start.py

```
@@ -1,19 +1,161 @@
+                        _stop_project(conflicting_project, verbose=verbose)
+                    console.print(
+                        f"[bold green]✓[/bold green] Stopped project '{conflicting_project}'"
+                    )
+                else:
+                    stderr_console.print(
+                        f"[bold red]Error:[/bold red] Cannot start '{project_name}' while '{conflicting_project}' is using required ports."
+                    )
+                    raise typer.Exit(code=1)
+        except KeyboardInterrupt:
+            stderr_console.print("\n[yellow]Operation cancelled.[/yellow]")
+            raise typer.Exit(code=0)
```

Ruff (0.14.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)